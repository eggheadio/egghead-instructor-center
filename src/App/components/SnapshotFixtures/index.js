import React from 'react'

export const StringFixture = 'String'

export const ImageUrlFixture = 'https://s3.amazonaws.com/uifaces/faces/twitter/kimcool/128.jpg'

export const ComponentFixture = () => (
  <div>
    Component
  </div>
)
