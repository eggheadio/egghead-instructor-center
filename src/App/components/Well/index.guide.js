import React from 'react'
import {map, keys} from 'lodash'
import {storiesOf} from '@kadira/storybook'
import {lorem} from 'faker'
import GuideVariation from '../GuideVariation'
import Well, {types} from '.'

storiesOf('Well', module)

  .add('types', () => (
    <div>
      {map(keys(types), type => (
        <GuideVariation
          key={type}
          tooltip={type}
        >
          <Well type={type}>
            {lorem.text()}
          </Well>
        </GuideVariation>
      ))}
    </div>
  ))
