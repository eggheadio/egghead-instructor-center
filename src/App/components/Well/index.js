import React from 'react'

export const types = {
  info: 'bg-light-blue',
  error: 'bg-light-red',
}

export default ({
  children,
  type = 'info',
}) => (
  <div className={`pa3 br2 ${types[type]}`}>
    {children}
  </div>
)
