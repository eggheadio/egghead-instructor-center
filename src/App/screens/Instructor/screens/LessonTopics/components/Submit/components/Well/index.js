import React from 'react'

const typeSpecificClassNames = {
  info: 'bg-light-blue',
  error: 'bg-light-red',
}

export default ({
  children,
  type = 'info',
}) => (
  <div className={`pa3 br2 ${typeSpecificClassNames[type]}`}>
    {children}
  </div>
)
