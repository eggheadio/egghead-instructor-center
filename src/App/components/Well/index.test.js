import React from 'react'
import {create} from 'react-test-renderer'
import {forEach, keys} from 'lodash'
import {StringFixture} from '../SnapshotFixtures'
import Well, {types} from '.'

test('types', () => {
  forEach(keys(types), (type) => (
    expect(create(
      <Well type={type}>
        {StringFixture}
      </Well>
    )).toMatchSnapshot()
  ))
})

