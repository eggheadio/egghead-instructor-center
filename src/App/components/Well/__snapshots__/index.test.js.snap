exports[`test types 1`] = `
<div
  className="pa3 br2 bg-light-blue">
  String
</div>
`;

exports[`test types 2`] = `
<div
  className="pa3 br2 bg-light-red">
  String
</div>
`;
